File: examples/spring-io-2026/java/src/main/java/org/example/koog/spring/java/structs/TransferResult.java
package org.example.koog.spring.java.structs;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonProperty;

import jakarta.annotation.Nullable;

public record TransferResult(
        @JsonProperty("was_successful") boolean wasSuccessful,
        @JsonProperty("problem") @Nullable String problem,
        @JsonProperty("transaction_id") @Nullable String transactionId
) {
    @JsonCreator
    public TransferResult {
    }
}

File: examples/spring-io-2026/kotlin/src/main/kotlin/org/example/koog/spring/kotlin/structs/TransferResult.kt
package org.example.koog.spring.kotlin.structs

import com.fasterxml.jackson.annotation.JsonCreator
import kotlinx.serialization.Serializable

@Serializable
data class TransferResult @JsonCreator constructor(
    val wasSuccessful: Boolean,
    val problem: String,
    val transactionId: String
) 